app/controllers/Application.java

```diff
@@ -1869,6 +1886,88 @@ public static Result getJobBestParameter(String jobDefId) {
     bestParamValueMap.put("executionId", bestTuningJobExecutionParamSet.jobExecution.jobExecUrl);
     return ok(Json.toJson(bestParamValueMap));
   }
+
+  /** Rest Api implementation for authenticating the user using Scheduler's Authentication API
+   * @result JSON containing session.id on successful authentication else the error message
+   * eg: Formatted JSON Data
+   * {
+   *  "status" : "success"
+   *  "session_id" : "4b52bfe1-5a22-9a56-8ced-955dd5b9f0ea"
+   *}
+   **/
+  public static Result authenticateUser() {
+    DynamicForm form = Form.form().bindFromRequest(request());
+    final String username = form.get(USERNAME);
+    final String password = form.get(PASSWORD);
+    final String schedulerUrl = form.get(SCHEDULER_URL);
+    logger.info("Authenticating user: " + username + " " + schedulerUrl);
+    logger.info(isSet(username) + " " + isSet(password) + " " + isSet(schedulerUrl));
+    if (!isSet(username) || !isSet(password)) {
+      return badRequest("Username or Password cannot be empty");
+    } else if (!isSet(schedulerUrl)) {
+      return badRequest("Scheduler Url is empty!!");
+    }
+    Map<String, String> response;
+    try {
+      response = authenticateUser(username, password, schedulerUrl + AZKABAN_AUTHENTICATION_URL_SUFFIX);
+    } catch (Exception ex) {
+      logger.error("Some error occured while authenticating the user ", ex);
+      return internalServerError();
+    }
+    return ok(Json.toJson(response));
+  }
+
+  /** Rest Api implementation for providing the status of user's authorization to WRITE
+   * to a respective project of a Scheduler
+   *
+   * @param jobDefId - JobDefintion Id specific to the project
+   * @param schedulerUrl - The url of the scheduler to which the job belongs
+   * @param sessionId - the scheduler session_id of the user
+   * @result JSON containing a Boolean (hasWritePermission) which tells if
+   * the user has WRITE access to the project
+   * eg: Formatted JSON Data
+   * {
+   *  "hasWritePermission" : "true"
+   *}
+   **/
+  public static Result getUserAuthorizationStatus(String sessionId, String jobDefId, String schedulerUrl) {
+    if (!isSet(sessionId) || !isSet(jobDefId) || !isSet(schedulerUrl)) {
+      return badRequest("SessionId or JobDefId or SchedulerUrl cannot be empty");
+    }
+    Map<String, String> responseMap = new HashMap();
+    try {
+      Map<String, String> queryParams = new HashMap();
+      queryParams.put(AZKABAN_SESSION_ID_KEY, sessionId);
+      queryParams.put(AJAX, AUTHORIZATION_AJAX_ENDPOINT);
+      //Extracting Project Name from the Job Definition id
+      List<NameValuePair> jobDefQueryParams = URLEncodedUtils.parse(new java.net.URI(jobDefId), "UTF-8");
+      for (NameValuePair param: jobDefQueryParams) {
+        if (param.getName().equals(PROJECT_KEY)) {
+          logger.info(PROJECT_KEY + " " + param.getValue());
```

Comment:
can we make thin logger statement to debug

Reply:
This statement got left out, there is no need of this statement so removing this log.